pallets/settlement/src/lib.rs

```diff
@@ -1168,6 +1359,120 @@ impl<T: Trait> Module<T> {
+            for ticker in &tickers {
+                if Self::venue_filtering(ticker) && !Self::venue_allow_list(ticker, venue_id) {
+                    failed = true;
+                    Self::deposit_event(RawEvent::VenueUnauthorized(
+                        SettlementDID.as_id(),
+                        *ticker,
+                        venue_id,
+                    ));
+                }
+            }
+
+            if !failed {
+                let mut tx_data = Self::mercat_tx_data(instruction_id);
+                assert_eq!(tx_data.len(), 3); // TODO: replace with ensure and proper Error
+
+                // TODO: change the `MercatTxDataStorage` to make this type of query cleaner.
+                let tx_data = tx_data.remove(2);
+                let decoded_justified_tx = match tx_data {
+                    MercatTxData::JustifiedTransfer(finalized) => {
+                        let mut data: &[u8] = &finalized;
+                        JustifiedTransferTx::decode(&mut data).unwrap()
+                    }
+                    _ => {
+                        panic!("Unexpected data type"); // TODO replace with proper Error
+                    }
+                };
+
+                // Verify the proofs.
+                let (new_sender_balance, new_receiver_balance) = TransactionValidator {}
+                    .verify_transaction(
+                        decoded_justified_tx,
+                        sndr_account: &PubAccount, // TODO: Get the account id for the modified UserAuth, then get public account from Confidential Asset Pallet
+                        sndr_init_balance: &EncryptedAmount, // similar to the previous comment
+                        rcvr_account: &PubAccount, // similar to the previous comment
+                        rcvr_init_balance: &EncryptedAmount, // similar to the previous comment
+                        &[],
+                        &mut rng,
+                    )
+                    .unwrap(); // TODO remove unwrap.
+
+                match with_transaction(|| {
+                    for (leg_id, leg_details) in legs.iter().filter(|(leg_id, _)| {
+                        let status = Self::instruction_leg_status(instruction_id, leg_id);
+                        status == LegStatus::ExecutionPending
+                    }) {
+                        //let result = T::Asset::base_transfer(
+                        //    leg_details.from,
+                        //    leg_details.to,
+                        //    &leg_details.asset,
+                        //    leg_details.amount,
+                        //);
+                        // TODO: update the encrypted account balances
+                        if let Ok(post_info) = result {
+                            transaction_weight += post_info.actual_weight.unwrap_or_default();
+                        } else {
+                            return Err(leg_id);
+                        }
```

Comment:
Instead of `for`, a `try_for_each` would be better here.

Reply:
I don't see a `for` in the highlighted code section.

Reply:
The `for` is a few lines above.

Reply:
I see. But, that whole piece of code is a scary looking, nested monster. I'll try to fix it in another PR if you don't mind.

Reply:
This will be part of the same task as well @satyamakgec